app/src/main/java/com/realityexpander/tasky/agenda_feature/data/repositories/eventRepository/remote/eventApi/eventApiImpls/EventApiImpl.kt

```diff
@@ -138,6 +104,24 @@ class EventApiImpl @Inject constructor(
         }
     }
 
+    ////////////////// HELPER FUNCTIONS //////////////////
+
+    private fun getMultipartBodyPartsForLocalPhotos(
+        photos: List<PhotoDTO.Local>
+    ): List<MultipartBody.Part> {
+        return photos.mapIndexed { index, photo ->
+            val bytes = photo.uri.getBytesFromUri(context, UPLOAD_IMAGE_MAX_SIZE)
+                ?: throw Exception("Photo not found")
```

Comment:
Do you catch this somewhere? What if the user adds a photo, then deletes it from their device, goes back to the app and hits save?

Reply:
Good point, I will put some error handling in for this.

Reply:
## FOLLOW UP QUESTION:

I tested this case of deleted photo, and it currently gives a system level error that is coming from the `Uri` handling functions. 

I see 2 approaches:

1. When the user selects the photos, copy it to the app cache and then upload the file data from there and delete when uploaded.
2. When the user presses `save`, check the Uri's for validity and if all are not valid, show error dialog and then remove the photo(s) that are no longer valid.

I will implement #2 because its easier for now, but wondering if #1 is better long term.

Reply:
I think either is fine and each has its up- and downside. The first won't lead to errors, but increases app storage and complexity (also things like what if there's no space left, which might confuse a user since they'll think the photo is already saved on their device) while the latter is easier, but comes with an error. In the end it's really just an edge case and even if this was an app with lots of users maybe 1% of them would ever face this.

Reply:
OK, i went for Option 1, as its easier to implement, an edge case, and small impact. 

But at least the error is not cryptic, and the edge case is covered.

For more serious type of online data gathering, i would cache everything, just in case.